src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2397,15 +2395,18 @@ void MergeTreeData::clearPartsFromFilesystemImpl(const DataPartsVector & parts_t
     /// This flag disallow straightforward concurrent parts removal. It's required only in case
     /// when we have parts on zero-copy disk + at least some of them were mutated.
     bool remove_parts_in_order = false;
-    if (settings->allow_remote_fs_zero_copy_replication && dynamic_cast<StorageReplicatedMergeTree *>(this) != nullptr)
+    if (dynamic_cast<StorageReplicatedMergeTree *>(this) != nullptr)
     {
         remove_parts_in_order = std::any_of(
             parts_to_remove.begin(), parts_to_remove.end(),
-            [] (const auto & data_part) { return data_part->isStoredOnRemoteDiskWithZeroCopySupport() && data_part->info.getMutationVersion() > 0; }
+            [] (const auto & data_part)
+            {
+                return (data_part->isStoredOnRemoteDiskWithZeroCopySupport() || data_part->isStoredOnRemoteDisk())
```

Comment:
And `data_part->isStoredOnRemoteDiskWithZeroCopySupport()=>| data_part->isStoredOnRemoteDisk()`